server/src/test/scala/org/scassandra/server/actors/BatchHandlerTest.scala

```diff
@@ -103,4 +109,28 @@ class BatchHandlerTest extends TestKit(ActorSystem("BatchHandlerTest")) with Fun
       ), ONE, LOGGED))
     )
   }
+
+  test("Prime for read time out exception - prepared statement") {
+    val batchMessage: Array[Byte] = dropHeaderAndLength(createBatchMessage(
+      List(PreparedStatement(1)),
+      streamId))
+    val errorResult: ReadRequestTimeoutResult = ReadRequestTimeoutResult()
+    when(primeBatchStore.findPrime(any(classOf[BatchExecution]))).thenReturn(Some(BatchPrime(errorResult)))
+
+    underTest ! BatchHandler.Execute(ByteString(batchMessage), streamId)
+    prepareHandlerProbe.expectMsg(PreparedStatementQuery(List(1)))
+    prepareHandlerProbe.reply(PreparedStatementResponse(Map(1 -> "insert into something")))
+    connectionTestProbe.expectMsg(cqlMessageFactory.createErrorMessage(errorResult, streamId, ONE))
+  }
+
+  test("Prime for read time out exception - query") {
```

Comment:
Awesome - will do



> On 19 Aug 2015, at 00:09, Andrew Tolbert <notifications@github.com> wrote:
> 
> In server/src/test/scala/org/scassandra/server/actors/BatchHandlerTest.scala:
> 
> > @@ -103,4 +109,28 @@ class BatchHandlerTest extends TestKit(ActorSystem("BatchHandlerTest")) with Fun
> >        ), ONE, LOGGED))
> >      )
> >    }
> > +
> > +  test("Prime for read time out exception - prepared statement") {
> > +    val batchMessage: Array[Byte] = dropHeaderAndLength(createBatchMessage(
> > +      List(PreparedStatement(1)),
> > +      streamId))
> > +    val errorResult: ReadRequestTimeoutResult = ReadRequestTimeoutResult()
> > +    when(primeBatchStore.findPrime(any(classOf[BatchExecution]))).thenReturn(Some(BatchPrime(errorResult)))
> > +
> > +    underTest ! BatchHandler.Execute(ByteString(batchMessage), streamId)
> > +    prepareHandlerProbe.expectMsg(PreparedStatementQuery(List(1)))
> > +    prepareHandlerProbe.reply(PreparedStatementResponse(Map(1 -> "insert into something")))
> > +    connectionTestProbe.expectMsg(cqlMessageFactory.createErrorMessage(errorResult, streamId, ONE))
> > +  }
> > +
> > +  test("Prime for read time out exception - query") {
> If you mix-in ErrorHandlingBehaviors (like ExecuteHandlerTest and QueryHandlerTest do) you can have this test all the various error message types. Here's an implementation of executeWithError for BatchHandlerTest that would work:
> 
>   override def executeWithError(result: ErrorResult, expectedError: (Byte, Consistency) => Error): Unit = {
>     val batchMessage: Array[Byte] = dropHeaderAndLength(createBatchMessage(
>       List(SimpleQuery("super simple query")),
>       streamId))
>     when(primeBatchStore.findPrime(any(classOf[BatchExecution]))).thenReturn(Some(BatchPrime(result)))
> 
>     underTest ! BatchHandler.Execute(ByteString(batchMessage), streamId)
> 
>     connectionTestProbe.expectMsg(expectedError(streamId, ONE))
>   }
> For this to work you'd also have to change ErrorHandlingBehaviors to need a FunSuiteLike instead of a FunSuite.
> 
> —
> Reply to this email directly or view it on GitHub.
>